### ISSUE TITLE
mate-control-volume-applet is not proxyed by xembedsniproxy

### ISSUE THREAD

**Issue by ghostplant** [rilian-la-te/xfce4-sntray-plugin#25]
Why mate-control-volume-applet from mate-media is not proxyed? Is there any solution?

**Comment by rilian-la-te** [comment]
KDE proxy? If my proxy, it is not working now and disabled. It is just a bunch of code. When it will finished, I will remove xembed applet.

**Comment by rilian-la-te** [comment]
If KDE proxy, it is a bug and should be fixed. I will look for it then.

**Comment by ghostplant** [comment]
I have no idea about the name since the naming is a little mess.
I just install this branch to support new indicator for vala-panel under mate.

Originally, there are 3 indicators in legacy xembed: nm-applet (from network-manager-gnome), ibus (input method), and mate-volume-applet (from mate-media).
After adding this plugin extension and disable legacy xembed, I see nm-applet and ibus are successfully  displayed by xfce4-sntray-plugin with new style, but mate-volume-applet is not displayed in it, it can only show in legacy xembed currently.

Hope to get fixed soon!

**Comment by ghostplant** [comment]
Maybe it is not related to `xembedsniproxy`, I just mean that mate-volume-applet cannot displayed inside xfce4-sntray-plugin.

**Comment by rilian-la-te** [comment]
Yes, I know, because it uses xembed. But nm-applet and ibus on Ubuntu uses appindicator AFAIK, which purpose is to create SNI for Unity (Unity can use xembed, but only for whitelisted binary applications)

**Comment by ghostplant** [comment]
So is there any solution to support mate-volume-applet by xfce4-sntray-plugin?

**Comment by rilian-la-te** [comment]
I do not know. You can try KDE's xembedsniproxy, but I did not any tests with it. Or wait when I reimplement GTK proxy myself. Or you can develop it and make a PR. But why you do not like vala-panel-extras-volume?

**Comment by ghostplant** [comment]
vala-panel-extras-volume would be OK if it is more likely to be a native applet.

1) Not suitable to have Quit Item in popup menu;
2) Beep sound according to current sound theme when altering the volume;

Does the second requirement easily to satisfy?

**Comment by rilian-la-te** [comment]
Why not Quit? About beeping - I will think about it.

Native applet - why?

**Comment by rilian-la-te** [comment]
vala-panel-extras-volume can be used not only in MATE, but in all DEs. I can add beeping, but I do not want to have any MATE dependencies. Or GNOME. If it can be done without this dependencies - it can be done by me as option.

**Comment by rilian-la-te** [comment]
About sounds. It can be done via GSound (and will be).
But it has sound in notification already. But I do not know whether your notification daemon support sounds.
